schema/query/projection_evaluator.go

```diff
@@ -0,0 +1,170 @@
+package query
+
+import (
+	"context"
+	"fmt"
+
+	"sync"
+
+	"github.com/rs/rest-layer/schema"
+)
+
+// Eval evaluate the projection on the given payload with the help of the
+// validator. The resolver is used to fetch payload of references outside of the
+// provided payload.
+func (p Projection) Eval(ctx context.Context, payload map[string]interface{}, validator schema.Validator, resolver ReferenceResolver) (map[string]interface{}, error) {
```

Comment:
> Another idea:
I didn't see your comment when I posted my last-two comments btw. I will have a look at the last commit.